### DISCUSSION TITLE
Clicking on different DialogTriggers consecutively adds a new FluentProvider div to the DOM

### DISCUSSION THREAD

**Discussion by adrianpzamora** [microsoft/fluentui#31356] [Q&A] [answered]
I implemented a component that that is essentially a wrapper to any button. This component contains a Dialog in which the wrapped button becomes the DialogTrigger to a DialogSurface. I have a grid of tiles that is each using the wrapper component, and it is opening up the DialogSurface fine. However, when I inspect the DOM, I see that it is adding a new div every time the dialog is opened. How can I prevent this from happening?

**Comment by bsunderhus** [comment, marked as answer]
TLDR; You cannot 😅

Hey there @adrianpzamora. That extra node is a portal where the dialog will be rendered. that is a common strategy for surface elements, like popovers/dialog/menu and etc,. adding it to a different order in the DOM, right after the whole content makes rendering it above content easier (not many layouting recalc fights), it also makes keyboard navigation and tab indexing less nuanced, as it's the last element on the page. The same happens for our Popover, Menu, Dropdown, Combobox and Tooltip components.

Here's the line code that declares the Portal on the DialogSurface: 

https://github.com/microsoft/fluentui/blob/9c98ebd056e4eb85893e16ab7552a44b5396fb7e/packages/react-components/react-dialog/src/components/DialogSurface/renderDialogSurface.tsx#L16
